### PR TITLE
Correct production pool

### PR DESCRIPTION
**Pull request by aaronFranssell**
Set the correct Cognito user pool to be used throughout the SDC. Previously the user pool in use in production was the development user pool. This caused several issues around deployments and debugging.

### CODE REVIEW

File: webportal/webapp/angular.json [hunk 1]
```diff
@@ -45,6 +45,23 @@
                   "with": "src/environments/environment.prod.ts"
                 }
               ]
+            },
+            "test": {
```
Comment (aaronFranssell) [inline webportal/webapp/angular.json:49]: In order to facilitate testing for this change, I added the test environment.

File: webportal/webapp/src/app/account/loginsync/components/loginsync.component.ts [hunk 2]
```diff
@@ -27,18 +27,11 @@ export class LoginSyncComponent implements OnInit {
...
-            window.location.href = this.buildRedirectUrl();
+            window.location.href = this.cognitoService.buildLoginGovRedirectUrl();
           },
           error => {
             this.linkSuccessful = false;
             this.errorMessage =  error;
           });
   }
-
-  buildRedirectUrl(): string {
```
Comment (aaronFranssell) [inline webportal/webapp/src/app/account/loginsync/components/loginsync.component.ts:38]: I pushed this functionality out to the CognitoService to centralize it.

File: webportal/webapp/src/app/account/loginsync/services/loginsyncservice.service.ts [hunk 3]
```diff
@@ -25,7 +24,7 @@ export class LoginSyncService {
   userAccountsLinked(): Observable<any> {
     return this.http.get(this.accountLinkedUrl, this.httpOptions)
       .map((response) => {
-        return Observable.of(response);
```
Comment (aaronFranssell) [inline webportal/webapp/src/app/account/loginsync/services/loginsyncservice.service.ts:28]: Fun and interesting Angular/Typescript shenanigans... This code already returns an Observable. Wrapping it in an `Observable.of` creates a nested Observable.

File: webportal/webapp/src/environments/environment.prod.ts [hunk 4]
```diff
@@ -3,12 +3,13 @@ export const environment = {
   production: true,
   STREAMING_URL: 'https://stream.securedatacommons.com/guacamole/?authToken=',
   REGION: 'us-east-1',
-  USER_POOL_ID: 'us-east-1_Y5JI7ysvY',
-  CLIENT_ID: '7t034p7j0aklgipi5kk8ll5udd',
+  USER_POOL_ID: 'us-east-1_1zLqP2XrI',
```
Comment (aaronFranssell) [inline webportal/webapp/src/environments/environment.prod.ts:6]: The correct production pool, yay!

File: webportal/webapp/src/factories/window.factory.ts [hunk 5]
```diff
@@ -0,0 +1,4 @@
+import { InjectionToken } from '@angular/core';
```
Comment (aaronFranssell) [inline webportal/webapp/src/factories/window.factory.ts:1]: Stackoverflow told me this is the best way to inject `window`... In tests now, `window` can be mocked out.

File: webportal/webapp/src/app/account/loginsync/guards/loginsync.guard.spec.ts [hunk 6]
```diff
@@ -1,31 +1,100 @@
 import { LoginSyncGuard } from './loginsync.guard';
+import { CognitoService } from '../../../../services/cognito.service';
+import { Router } from '@angular/router';
```
Comment (TiannaW24) [inline webportal/webapp/src/app/account/loginsync/guards/loginsync.guard.spec.ts:3]: been meaning to get back to these tests for a while. thanks

File: webportal/webapp/src/app/account/loginsync/services/loginsyncservice.service.ts (hunk 3 above)
Comment (TiannaW24) [reply webportal/webapp/src/app/account/loginsync/services/loginsyncservice.service.ts:28]: Hmmm I specifically remember being mad about this and then getting a type error when I removed it...

File: webportal/webapp/src/environments/environment.prod.ts [hunk 7]
```diff
@@ -3,12 +3,13 @@ export const environment = {
...
   IDENTITY_PROVIDER: 'USDOT-ADFS',
-  APP_DOMAIN: 'dev-sdc-dot-webportal',
+  APP_DOMAIN: 'prod-sdc-dot-webportal',
   REDIRECT_URL: 'https://portal.securedatacommons.com/index.html',
-  API_ENDPOINT: 'https://cyfca70occ.execute-api.us-east-1.amazonaws.com/api/',
-  LOGIN_GOV_COGNITO_APP_CLIENT_ID: '34vj4387sna9jk31gudpt3b78i',
+  API_ENDPOINT: 'https://bzzlrrsxa4.execute-api.us-east-1.amazonaws.com/api/',
+  LOGIN_GOV_COGNITO_APP_CLIENT_ID: '3mlku34f58oeloqqc3hpiko6l3',
   LOGIN_GOV_ACCOUNT_LINK_URL: 'https://2l418womp5.execute-api.us-east-1.amazonaws.com',
+  ENVIRONMENT: 'prod',
```
Comment (TiannaW24) [inline webportal/webapp/src/environments/environment.prod.ts:14]: Do we need this? We have production: true set already.

File: webportal/webapp/angular.json (hunk 1 above)
Comment (TiannaW24) [reply webportal/webapp/angular.json:49]: Isn't that what dev is for?  Do we even have a working test cognito pool?

File: webportal/webapp/src/environments/environment.prod.ts (hunk 7 above)
Comment (TiannaW24) [reply webportal/webapp/src/environments/environment.prod.ts:14]: Oh, are you trying to differentiate between dev and test?

Comment (aaronFranssell) [reply webportal/webapp/src/environments/environment.prod.ts:14]: Yep. Previously with 2 environments asking if the environment was `prod` was enough. Now we actually have to know.

**Pull request merged**